src/app/util/binding-table.h

```diff
@@ -23,8 +23,67 @@
 
 #include <app/util/af-types.h>
 
-EmberStatus emberGetBinding(uint8_t index, EmberBindingTableEntry * result);
+namespace chip {
 
-EmberStatus emberSetBinding(uint8_t index, EmberBindingTableEntry * result);
+class BindingTable
+{
+    friend class Iterator;
 
-EmberStatus emberDeleteBinding(uint8_t index);
+public:
+    BindingTable();
+
+    class Iterator
+    {
+        friend class BindingTable;
+
+    public:
+        EmberBindingTableEntry & operator*() { return mTable->mBindingTable[mIndex]; }
+
+        const EmberBindingTableEntry & operator*() const { return mTable->mBindingTable[mIndex]; }
+
+        EmberBindingTableEntry * operator->() { return &(mTable->mBindingTable[mIndex]); }
+
+        const EmberBindingTableEntry * operator->() const { return &(mTable->mBindingTable[mIndex]); }
+
+        Iterator operator++() const;
```

Comment:
This shouldn't be const.  It's explicitly a state-mutating operation: it advances this iterator.